### PR TITLE
DSM new changes

### PR DESCRIPTION
**Pull request by pegahtah**

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts [hunk 1]
```diff
@@ -6,16 +6,20 @@ export class ActivityDefinition {
     public activityCode: string,
     public activityName: string,
     public activityVersion: string,
-    public questions: Array<QuestionDefinition>
+    public questions: Array<QuestionDefinition>,
+    public displayOrder:number=1, public showActivityStatus = true
```
Comment (SimoneMaiwald) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts]: can we use the new values from activity_definition

File: ddp-workspace/projects/ddp-dsm-ui/src/app/filter-column/filter-column.model.ts [hunk 2]
```diff
@@ -239,6 +242,26 @@ export class Filter {
     new NameValue('review', 'Second Abstraction'),
     new NameValue('qc', 'QC') ]);
   public static ABSTRACTION_USER = new Filter(ParticipantColumn.ABSTRACTION_USER, Filter.TEXT_TYPE);
+  public static ACTIVITY_STATUS = new Filter(ParticipantColumn.ACTIVITY_STATUS, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null,
+    (participant: Participant,  activityDefinitionList: ActivityDefinition[])=>{
+      let str = "";
+      let activityDataArray = participant.data.activities;
```
Comment (SimoneMaiwald) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/filter-column/filter-column.model.ts]: why do we need participant.data.activities here. do we not have all info in activityDefinitionList?

Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/filter-column/filter-column.model.ts]: we want to sort activities in the order that was in their definition

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts (hunk 1 above)
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts]: I just added them as default values if the json doesn't have them but the column is selected

File: ddp-workspace/projects/ddp-dsm-ui/src/app/tissue-list/tissue-list.component.ts [hunk 3]
```diff
@@ -1660,12 +1665,59 @@ export class TissueListComponent implements OnInit {
...
+    let assigneesMap = this.getAssigneeAsNameValue();
+    let selectedOptions = filter.getSelectedOptionsBoolean( assigneesMap );
+    filter.selectedOptions = selectedOptions;
+    let f = this.selectedColumns[ 'data' ].find( filter => filter.participantColumn.name === ParticipantColumn.ASSIGNEE_TISSUE.name );
+    if (f) {
+      let index = this.selectedColumns[ 'data' ].indexOf( f );
+      this.selectedColumns[ 'data' ].splice( index, 1 );
+      f.selectedOptions = selectedOptions;
+      this.selectedColumns[ 'data' ].push( f );
+      for (let data of this.dataSources) {
```
Comment (SimoneMaiwald) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/tissue-list/tissue-list.component.ts]: maybe we should rename that data to something else. makes it hard to read with the 'data' just a line up… maybe change it to source or dataSource

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts (hunk 1 above)
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts]: @SimoneMaiwald If you think unnecessary I can remove

Comment (SimoneMaiwald) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts]: if the json doesn't have them then default for showActivityStatus should be false

File: ddp-workspace/projects/ddp-dsm-ui/src/app/filter-column/filter-column.model.ts (hunk 2 above)
Comment (SimoneMaiwald) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/filter-column/filter-column.model.ts]: yes, but that order is in activity definition. why do we need participant object?

**Comment by alextanlan** [comment]
Can you please fix linter warnings ? 
run `ng lint ddp-dsm-ui` in ddp-workspace folder -> ✖ 84 problems (81 errors, 3 warnings)

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts (hunk 1 above)
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/models/activity-definition.model.ts]: fixed

File: ddp-workspace/projects/ddp-dsm-ui/src/app/filter-column/filter-column.model.ts (hunk 2 above)
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/filter-column/filter-column.model.ts]: basically here I am sorting that participant's activities based on their order, which is in the definition. So I would need the participant to access their activities, and the definition to sort them

File: ddp-workspace/projects/ddp-dsm-ui/src/app/tissue-list/tissue-list.component.ts (hunk 3 above)
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/tissue-list/tissue-list.component.ts]: fixed

**Comment by pegahtah** [comment]
> Can you please fix linter warnings ? run `ng lint ddp-dsm-ui` in ddp-workspace folder -> ✖ 84 problems (81 errors, 3 warnings)

@alextanlan thank you for letting me know how to do it, all fixed

**Review by alextanlan — DISMISSED** [review]

**Review by SimoneMaiwald — DISMISSED** [review]

**Review by alextanlan — APPROVED** [review]

**Pull request merged**